### DISCUSSION TITLE
Changelog only with changes?

### DISCUSSION THREAD

**Discussion by Mordecaine** [netbox-community/netbox#6464] [Q&A] [answered]
Hello,

At this moment I sent updates to my netbox via API every 10 minutes. But in the most cases there is no change in my update (its like a sync job). The changelogs are now getting bigger and bigger and there is no useful way to find changes where something really changed.

Here is a small example:
```
address_obj.update({
            "vrf": vrf_obj.id,
            "tenant": tenant_obj.id,
            "status": "active",
            "description": compute.name,
            "assigned_object_id": nic_obj.id,
            "tags": ip_address_tags,
            "assigned_object_type": "virtualization.vminterface"
        })
```
But if there is no change in it, it will create a changelog entry with no changes:

![netbox_question](https://user-images.githubusercontent.com/25454900/119139116-f2db5380-ba42-11eb-96d6-bafb9f47660c.jpg)

Now my questions:

1. Is there a way to get only the changes in the changelog where something changed?
2. Is there a way to delete all changelogs without changes?

**Comment by ryanmerolle** [comment, marked as answer]
At least in 3.0 you can filter in the gui changelogs and delete accordingly.  You can also as always query via the api.

**Reply by candlerb** [reply]
This was raised as an issue: #6599
But @jeremystretch said he wanted to record every save, even if it hadn't changed anything.
